memcached/src/main/java/com/yahoo/ycsb/db/MemcachedCompatibleClient.java

```diff
@@ -0,0 +1,239 @@
+    try {
+      client = createMemcachedClient();
+      checkOperationStatus = Boolean.parseBoolean(
+          getProperties().getProperty(CHECK_OPERATION_STATUS_PROPERTY,
+                                      CHECK_OPERATION_STATUS_DEFAULT));
+      objectExpirationTime = Integer.parseInt(
+          getProperties().getProperty(OBJECT_EXPIRATION_TIME_PROPERTY,
+                                      DEFAULT_OBJECT_EXPIRATION_TIME));
+      shutdownTimeoutMillis = Integer.parseInt(
+          getProperties().getProperty(SHUTDOWN_TIMEOUT_MILLIS_PROPERTY,
+                                      DEFAULT_SHUTDOWN_TIMEOUT_MILLIS));
+    } catch (Exception e) {
+      throw new DBException(e);
+    }
+  }
+
+  protected abstract MemcachedClient createMemcachedClient() throws Exception;
+
+  @Override
+  public Status read(
+      String table, String key, Set<String> fields,
+      HashMap<String, ByteIterator> result) {
+    key = createQualifiedKey(table, key);
+    try {
+      GetFuture<Object> future = memcachedClient().asyncGet(key);
+      Object document = future.get();
+      if (document != null) {
+        fromJson((String) document, fields, result);
+      }
+      return Status.OK;
+    } catch (Exception e) {
+      logger.error("Error encountered for key: " + key, e);
+      return new Status("ERROR_READ", e.toString());
+    }
+  }
+
+  @Override
+  public Status scan(
+      String table, String startkey, int recordcount, Set<String> fields,
+      Vector<HashMap<String, ByteIterator>> result){
+    return Status.NOT_IMPLEMENTED;
+  }
+
+  @Override
+  public Status update(
+      String table, String key, HashMap<String, ByteIterator> values) {
+    key = createQualifiedKey(table, key);
+    try {
+      OperationFuture<Boolean> future =
+          memcachedClient().replace(key, objectExpirationTime, toJson(values));
+      return getReturnCode(future);
+    } catch (Exception e) {
+      logger.error("Error updating value with key: " + key, e);
+      return new Status("ERROR_UPDATE", e.toString());
+    }
+  }
+
+  @Override
+  public Status insert(
+      String table, String key, HashMap<String, ByteIterator> values) {
+    key = createQualifiedKey(table, key);
+    try {
+      OperationFuture<Boolean> future =
+          memcachedClient().add(key, objectExpirationTime, toJson(values));
+      return getReturnCode(future);
+    } catch (Exception e) {
+      logger.error("Error inserting value", e);
+      return new Status("ERROR_INSERT", e.toString());
+    }
+  }
+
+  @Override
+  public Status delete(String table, String key) {
+    key = createQualifiedKey(table, key);
+    try {
+      OperationFuture<Boolean> future = memcachedClient().delete(key);
+      return getReturnCode(future);
+    } catch (Exception e) {
+      logger.error("Error deleting value", e);
+      return new Status("ERROR_DELETE", e.toString());
+    }
+  }
+
+  public int query(String table, String key, int limit) {
```

Comment:
I do think it would be interesting to see query get implemented in some way in the YCSB framework. Currently it is not implemented, can this method be removed?